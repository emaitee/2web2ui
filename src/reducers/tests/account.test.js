import cases from 'jest-in-case';
import accountReducer, { initialState } from '../account';

const CREATE_TEST_CASES = {
  'create account pending': {
    type: 'CREATE_ACCOUNT_PENDING'
  },
  'create account fail': {
    type: 'CREATE_ACCOUNT_FAIL',
    payload: { errors: [{ message: 'Some error occurred' }]}
  },
  'create account success': {
    type: 'CREATE_ACCOUNT_SUCCESS',
    payload: {
      'results': {
        'message': 'Account created, welcome to SparkPost!',
        'username': 'john.smith',
        'customer_id': 1000242
      }
    }
  }
};

cases('Account reducer', (action) => {
  expect(accountReducer(initialState, action)).toMatchSnapshot();
}, CREATE_TEST_CASES);
