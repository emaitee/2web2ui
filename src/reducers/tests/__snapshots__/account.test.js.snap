// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account reducer create account fail 1`] = `
Object {
  "createError": Object {
    "errors": Array [
      Object {
        "message": "Some error occurred",
      },
    ],
  },
  "createLoading": false,
  "loading": false,
  "subscription": Object {},
  "updateError": null,
  "updateLoading": false,
}
`;

exports[`Account reducer create account pending 1`] = `
Object {
  "createError": null,
  "createLoading": true,
  "loading": false,
  "subscription": Object {},
  "updateError": null,
  "updateLoading": false,
}
`;

exports[`Account reducer create account success 1`] = `
Object {
  "createLoading": false,
  "loading": false,
  "payload": Object {
    "results": Object {
      "customer_id": 1000242,
      "message": "Account created, welcome to SparkPost!",
      "username": "john.smith",
    },
  },
  "subscription": Object {},
  "updateError": null,
  "updateLoading": false,
}
`;
